rest/route/middleware.go

```diff
@@ -493,7 +496,7 @@ func (m *TaskAuthMiddleware) ServeHTTP(rw http.ResponseWriter, r *http.Request,
 		}))
 		return
 	}
-	if utility.StringSliceContains(evergreen.TaskCompletedStatuses, t.Status) {
+	if t.FinishTime.Before(time.Now().Add(-1*completedTaskValidityWindow)) && utility.StringSliceContains(evergreen.TaskCompletedStatuses, t.Status) {
```

Comment:
nit: I think I've seen some go linters say to use `time.Since`/`time.Until` when comparing against now, like `time.Since(t.FinishTime) > completedTaskValidityWindow`.